server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1661,6 +1661,19 @@ private boolean upgradeRunningVirtualMachine(Long vmId, Long newServiceOfferingI
                     + ",memory=," + currentMemory + ")");
         }
 
+        _offeringDao.loadDetails(currentServiceOffering);
```

Comment:
@nitin-maharana This check will be redundant as there is no place where details are getting set in variable's lifetime.